### PR TITLE
Improved Tests for Learning Exercise "The Farm"

### PR DESCRIPTION
**Pull request by saurabhmshr**
Improved case structure in test function `TestDivideFood`

Changes expected error message type from `string` to `error` in test function `TestDivideFoodSillyNephewError`

### CODE REVIEW

**Review by SR8H — COMMENTED** [review]
removed spaces for error variable

**Review by SR8H — COMMENTED** [review]
applied go fmt to file

File: exercises/concept/the-farm/the_farm_test.go [hunk 1]
```diff
@@ -14,192 +14,141 @@ func (wf testWeightFodder) FodderAmount() (float64, error) {
...
-					test.wantErrMsg,
-				)
-			}
-		})
+		t.Logf("Testing for %s", test.description)
+
+		fodderWeight := testWeightFodder{fodder: 100, err: nil}
+		gotFodderAmount, gotErrMsg := DivideFood(fodderWeight, test.cows)
+
+		if gotFodderAmount != 0 && gotErrMsg != test.wantErrMsg {
```
Comment (junedev) [inline exercises/concept/the-farm/the_farm_test.go:149]: Errors cannot be compared in this way, see https://go.dev/play/p/rRmtE4fDdQ9
Also, we don't expect a normal sentinel error here, we expect a special type of error that the student should write.

File: exercises/concept/the-farm/the_farm_test.go [hunk 2]
```diff
@@ -14,192 +14,141 @@ func (wf testWeightFodder) FodderAmount() (float64, error) {
...
+			description:      "100 fodder for zero cows with no error",
+			fodderAmount:     testWeightFodder{fodder: 100, err: nil},
+			cows:             0,
+			wantFodderAmount: 0,
+			wantFodderErr:    errors.New("division by zero"),
 		},
 	}
+
 	for _, test := range tests {
-		t.Run(test.description, func(t *testing.T) {
```
Comment (junedev) [inline exercises/concept/the-farm/the_farm_test.go:125]: Why should t.Run (a good practice in Go) be removed in favor of a self-written t.Logf?

File: exercises/concept/the-farm/the_farm_test.go [hunk 3]
```diff
@@ -14,192 +14,141 @@ func (wf testWeightFodder) FodderAmount() (float64, error) {
...
 	}{
 		{
-			description:             "100 fodder for 10 cows",
-			weightFodder:            testWeightFodder{fodder: 100, err: nil},
-			weightFodderDescription: "100 fodder, no error",
-			cows:                    10,
-			wantAmount:              10,
-			wantErr:                 nil,
+			description:      "100 fodder for 10 cows with no error",
+			fodderAmount:     testWeightFodder{fodder: 100, err: nil},
```
Comment (junedev) [inline exercises/concept/the-farm/the_farm_test.go:29]: This field contains a fodder, not an amount.

File: exercises/concept/the-farm/the_farm_test.go [hunk 4]
```diff
@@ -14,192 +14,141 @@ func (wf testWeightFodder) FodderAmount() (float64, error) {
 	return wf.fodder, wf.err
 }
 
+var NonScaleError = errors.New("non-scale error")
```
Comment (junedev) [inline exercises/concept/the-farm/the_farm_test.go:17]: This error was intenionally local to the test case as it is not relevant for other tests.

**Review by junedev — COMMENTED** [review]
I am sorry but this PR does not improve the exercise for the reasons mentioned in the comments. I will close it for now. Please open an issue describing the problem and the improvement you suggest and we can take it from there.

Keep the comment I made here in mind: https://github.com/exercism/go/issues/2477
If there are very important, concrete fixes that you think should be done, feel free to raise the issue. Otherwise, let's tackle this more generally as explained in the other issue.

File: exercises/concept/the-farm/the_farm_test.go (hunk 2 above)
Comment (saurabh-mish) [reply exercises/concept/the-farm/the_farm_test.go:125]: Running subtests with `t.Run` is a good practice. Its also a good practice to log tests and subtests. 

I can change the tests to have subtests only (`t.Run`)

File: exercises/concept/the-farm/the_farm_test.go (hunk 1 above)
Comment (saurabh-mish) [reply exercises/concept/the-farm/the_farm_test.go:149]: The special type of error that the student should write is already a part of `DivideFood`. This method returns the amount of fodder and error message. A combination of these return values are being compared with expected values from test data.

File: exercises/concept/the-farm/the_farm_test.go (hunk 3 above)
Comment (saurabh-mish) [reply exercises/concept/the-farm/the_farm_test.go:29]: Do you think its a good idea to update `wantAmount` and `wantErr` to `fodderAmount` and `fodderError` ?

File: exercises/concept/the-farm/the_farm_test.go (hunk 4 above)
Comment (saurabh-mish) [reply exercises/concept/the-farm/the_farm_test.go:17]: Agree

**Pull request closed without merging**